listings/views.py

```diff
@@ -2,14 +2,15 @@
 
 from rest_framework import viewsets
 from rest_framework.permissions import IsAuthenticated
-from rest_framework.response import Response
 
+from listings.filters import ListingFilter
 from listings.models import Listing
 from listings.serializers import ListingSerializer
 
 
 class ListingViewSet(viewsets.ModelViewSet):
     """ViewSet for the Listing model."""
-    queryset = Listing.objects.all().prefetch_related('images')
+    queryset = Listing.objects.all().prefetch_related('images').distinct()
     serializer_class = ListingSerializer
     permission_classes = [IsAuthenticated]
+    filter_class = ListingFilter
```

Comment:
shouldn't it be `filterset_class`? I've been looking at both
https://www.django-rest-framework.org/api-guide/filtering/#djangofilterbackend and
https://django-filter.readthedocs.io/en/stable/guide/rest_framework.html

Reply:
Yeah, old name. I'll change this.